docs/advanced.md

```diff
@@ -57,6 +57,28 @@ dispatch = httpx.WSGIDispatch(app=app, remote_addr="1.2.3.4")
 client = httpx.Client(dispatch=dispatch)
 ```
 
+## Build Request
+
+Using a Client instance to make build requests
```

Comment:
We should probably show the use-case here such as being able to modify the request in ways you usually can't with our standard interface. An example of this is making an `OPTIONS *` request for CORS:

```python
client = httpx.Client()
req = client.build_request("OPTIONS", "https://example.com")
req.url.full_path = "*"
resp = client.send(req)
```

(I don't think you can *currently* do this but you'll be able to for sure after my proxy PR lands)

Reply:
In your example we can't change `full_path`  because it's an attribute
Do you have any suggestions?

Reply:
You can't change it now but once another PR is merged you will be able to.
You shouldn't create a test case with it, but you can add it to documentation. :)